mne/preprocessing/tests/test_xdawn.py

```diff
@@ -262,4 +266,98 @@ def test_XdawnTransformer():
+    epoch_data[labels == 0, informative_ch_idx, :] += nontarget_template
+    epoch_data[labels == 1, informative_ch_idx, :] += target_template
+
+    mixing_mat = np.linalg.svd(rs.randn(n_channels, n_channels))[0]
+    mixed_epoch_data = np.dot(mixing_mat.T, epoch_data).transpose((1, 0, 2))
+
+    events = np.zeros((n_epochs, 3), dtype=int)
+    events[:, 0] = np.arange(0, n_epochs * n_times, n_times)
+    events[:, 2] = labels
+
+    info = create_info(
+        ch_names=['C{:02d}'.format(i) for i in range(n_channels)],
+        ch_types=['eeg'] * n_channels,
+        sfreq=sfreq)
+    epochs = EpochsArray(mixed_epoch_data, info, events,
+                         tmin=epoch_tmin,
+                         event_id={'nt': 0, 't': 1})
+
+    return epochs, mixing_mat
+
+
+@requires_sklearn
+def test_xdawn_decoding_performance():
+    """Test decoding performance and extracted pattern on synthetic data."""
+    from sklearn.model_selection import KFold
+    from sklearn.pipeline import make_pipeline
+    from sklearn.linear_model import LogisticRegression
+    from sklearn.preprocessing import MinMaxScaler
+    from sklearn.metrics import accuracy_score
+
+    n_xdawn_comps = 3
+    expected_accuracy = 0.85
+
+    epochs, mixing_mat = _simulate_erplike_mixed_data(n_epochs=100)
+    labels = epochs.events[:, 2]
+
+    # results of Xdawn and _XdawnTransformer should match
+    xdawn_pipe = make_pipeline(
+        Xdawn(n_components=n_xdawn_comps),
+        Vectorizer(),
+        MinMaxScaler(),
+        LogisticRegression(penalty='l1', solver='liblinear'))
+    xdawn_trans_pipe = make_pipeline(
+        _XdawnTransformer(n_components=n_xdawn_comps),
+        Vectorizer(),
+        MinMaxScaler(),
+        LogisticRegression(penalty='l1', solver='liblinear'))
+
+    for cur_pipeline, cur_pipeline_input in (
+            (xdawn_pipe, epochs),
+            (xdawn_trans_pipe, epochs.get_data())):
+        cv = KFold(n_splits=3, shuffle=False)
+        predictions = np.empty_like(labels, dtype=float)
+        for cur_train_idxs, cur_test_idxs in cv.split(cur_pipeline_input,
+                                                      labels):
+            cur_pipeline.fit(cur_pipeline_input[cur_train_idxs],
+                             labels[cur_train_idxs])
+            predictions[cur_test_idxs] = cur_pipeline.predict(
+                cur_pipeline_input[cur_test_idxs])
+
+        cv_accuracy_xdawn = accuracy_score(labels, predictions)
+        assert_allclose(cv_accuracy_xdawn, expected_accuracy, atol=0.01)
+
+        # for both event types, the first component should "match" the mixing
+        fitted_xdawn = cur_pipeline.steps[0][1]
+        if isinstance(fitted_xdawn, Xdawn):
+            relev_patterns = np.concatenate([
+                fitted_xdawn.patterns_[cur_class][:1, :]
+                for cur_class in epochs.event_id.keys()])
+        else:
+            relev_patterns = fitted_xdawn.patterns_[::n_xdawn_comps]
```

Comment:
::n_xdawn_comps -> :n_xdawn_comps ?

Reply:
This is as intended (we want the first of ``the n_xdawn_comps`` components  *per class*). We compare with the same pattern in ``mixing_mat`` later because the same original channel in the generated data has discriminative information for both classes.